### ISSUE TITLE
Example in README should show arguments with `isDue()` and README API documentation.

### ISSUE THREAD

**Issue by adjenks** [dragonmantank/cron-expression#59]
Initially I didn't know that `isDue()` could take parameters, so I made the mistake of calling it multiple times between potentially long running tasks, therefore offsetting the time because the function internally creates a new DateTime() object whenever it is called.

Because the README provides a "Usage" section and I was not using a code completion tool in my editor, I assumed that the entire API was demonstrated in the example. I later looked in the code to read the API and realised that there was more to it and that it was documented.

So I believe that either: The API should be fully documented in the README _or_ the README should demonstrate the use of the `isDue()` parameters and the README should inform users of where to look for API documentation in the code.

**Comment by dragonmantank** [comment]
So to understand your use case, you were starting a script, creating an expression, and then using that same expression multiple times throughout a long process, with the expectation that `isDue()` would always compare to the time of object creation, versus the time of method invocation?

I can look into setting up something to turn the docblocks into documentation, but as you noticed the code is always the best source of information. I feel that the README should continue to show the most common use cases and uses without becoming a massive manual.

**Comment by adjenks** [comment]
Yes, I used the function at first without thinking, then I looked through the README wondering how to anchor it to a particular time and couldn't see how. So I searched the code for the reference to the function and found that it could accept a DateTime object.

When possible, I'd rather just see the interface that a library exposes, instead of inspecting its inner workings. Most libraries I use provide API documentation somewhere in the README, or linked from the README, or simply by telling you which files contain the bulk of the API. Sometimes their API is hosted on another website when it is very elaborate.

This package has a very large number of downloads so I think the documentation is very important.
As you suggested, the docblocks could be converted into full documentation, then the README could simply point to the docs folder to keep it clean.
